plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -696,6 +697,20 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
                 .iterator();
     }
 
+    @Override
+    public Iterator<TableCommentMetadata> streamRelationComments(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        if (prefix.getTable().isPresent()) {
+            // For single name referenced the default implementation is good enough
+            return ConnectorMetadata.super.streamRelationComments(session, prefix);
```

Comment:
> Similarly, do we have a test that exercises the path where both schema and table are provided in the prefix?

adding test for this case, search for `// Pointed lookup`.

> but only because of caching.

i didn't want to optimize this path as i haven't seen this in the wild.

I couldn't get rid of reliance on caching because of how GlueCatalog and GlueTableOperations interact, so even the optimized path had this unfortunate dependency on the cache.

i don't want to pass prefix's down to the TrinoGlueCatalog implementation, to save complexity at least here. the biggest downside of this approach is complexity. some of the complexity seems unavoidable...

Reply:
i decided to go into opposite direction. see `adjust iceberg to Fixup simplify streamRelationComments: take Optional<String> schemaName instead of SchemaTablePrefix` commit. I don't think `tablePrefix` abstraction buys us anything, and it just pushes more complexity into connectors.